cognite/neat/graph/stores/_base.py

```diff
@@ -1,291 +1,165 @@
                 )
             ]
         )
+        self.rules = rules
 
-    @abstractmethod
-    def _set_graph(self) -> None:
-        raise NotImplementedError()
-
-    def init_graph(
-        self,
-        rdf_store_query_url: str | None = None,
-        rdf_store_update_url: str | None = None,
-        graph_name: str | None = None,
-        base_prefix: str | None = None,
-        returnFormat: str = "csv",
-        internal_storage_dir: Path | None = None,
-    ):
-        """Initializes the graph.
-
-        Args:
-            rdf_store_query_url : URL towards which SPARQL query is executed, by default None
-            rdf_store_update_url : URL towards which SPARQL update is executed, by default None
-            graph_name : Name of graph, by default None
-            base_prefix : Base prefix for graph namespace to change if needed, by default None
-            returnFormat : Transport format of graph data between, by default "csv"
-            internal_storage_dir : Path to directory where internal storage is located,
-                                   by default None (in-memory storage).
-
-        !!! note "internal_storage_dir"
-            Used only for Oxigraph
-        """
-        logging.info("Initializing NeatGraphStore")
-        self.rdf_store_query_url = rdf_store_query_url
-        self.rdf_store_update_url = rdf_store_update_url
-        self.graph_name = graph_name
-        self.returnFormat = returnFormat
-        self.internal_storage_dir = Path(internal_storage_dir) if internal_storage_dir else None
-        self.internal_storage_dir_orig = (
-            self.internal_storage_dir if self.internal_storage_dir_orig is None else self.internal_storage_dir_orig
-        )
+        if self.rules:
+            self._upsert_prefixes(self.provenance, self.graph, self.rules)
 
-        self._set_graph()
-
-        if self.prefixes:
-            for prefix, namespace in self.prefixes.items():
-                logging.info("Adding prefix %s with namespace %s", prefix, namespace)
-                self.graph.bind(prefix, namespace)
-
-        if base_prefix:
-            self.base_prefix = base_prefix
-        if self.base_prefix:
-            self.graph.bind(self.base_prefix, self.namespace)
-            logging.info("Adding prefix %s with namespace %s", self.base_prefix, self.namespace)
-        logging.info("Graph initialized")
-
-    def reinitialize_graph(self):
-        """Reinitialize the graph."""
-        self.init_graph(
-            self.rdf_store_query_url,
-            self.rdf_store_update_url,
-            self.graph_name,
-            self.base_prefix,
-            self.returnFormat,
-            self.internal_storage_dir,
-        )
+        self.queries = _Queries(self)
 
-    def upsert_prefixes(self, prefixes: dict[str, Namespace]) -> None:
+    @classmethod
+    def _upsert_prefixes(cls, provenance: Provenance, graph: Graph, rules: InformationRules) -> None:
```

Comment:
Any reason not to just make this a regular method?
```suggestion
    def _upsert_prefixes(self, prefixes: Iterable[Namespace]):-> None:
```